traffic_ops/testing/api/v3/deliveryservice_requests_test.go

```diff
@@ -16,366 +16,317 @@ package v3
-		XMLID := "X " + strings.Repeat("X", 46)
-		displayName := strings.Repeat("X", 49)
-
-		dsr := testData.DeliveryServiceRequests[dsrGood]
-		dsr.DeliveryService.DisplayName = displayName
-		dsr.DeliveryService.RoutingName = routingName
-		dsr.DeliveryService.XMLID = XMLID
-
-		_, _, err := TOSession.CreateDeliveryServiceRequest(dsr)
-		if err == nil {
-			t.Error("expected: validation error, actual: nil")
-		}
-	})
-}
-
-func TestDeliveryServiceRequestTypeFields(t *testing.T) {
-	WithObjs(t, []TCObj{CDNs, Types, Tenants, Parameters}, func() {
-		dsr := testData.DeliveryServiceRequests[dsrBadTenant]
-
-		alerts, _, err := TOSession.CreateDeliveryServiceRequest(dsr)
-		if err != nil {
-			t.Errorf("Error occurred %v", err)
-		}
-
-		found := false
-		for _, alert := range alerts.Alerts {
-			if alert.Level == tc.ErrorLevel.String() {
-				t.Errorf("Expected only succuss-level alerts creating a DSR, got error-level alert: %s", alert.Text)
-			} else if alert.Level == tc.SuccessLevel.String() {
-				t.Logf("Got expected alert creating a DSR: %s", alert.Text)
-				found = true
-			}
-		}
-		if !found {
-			t.Errorf("Expected a success-level alert creating a DSR, got none")
+		for method, testCases := range methodTests {
+			t.Run(method, func(t *testing.T) {
+				for name, testCase := range testCases {
+					dsReq := tc.DeliveryServiceRequest{}
+
+					if testCase.RequestBody != nil {
+						dat, err := json.Marshal(testCase.RequestBody)
+						assert.NoError(t, err, "Error occurred when marshalling request body: %v", err)
+						err = json.Unmarshal(dat, &dsReq)
+						assert.NoError(t, err, "Error occurred when unmarshalling request body: %v", err)
+					}
+
+					switch method {
+					case "GET", "GET AFTER CHANGES":
+						t.Run(name, func(t *testing.T) {
+							if name == "OK when VALID XMLID parameter" {
+								resp, reqInf, err := testCase.ClientSession.GetDeliveryServiceRequestByXMLIDWithHdr(testCase.RequestParams["xmlId"][0], testCase.RequestHeaders)
+								for _, check := range testCase.Expectations {
+									check(t, reqInf, resp, tc.Alerts{}, err)
+								}
+							} else {
+								resp, reqInf, err := testCase.ClientSession.GetDeliveryServiceRequestsWithHdr(testCase.RequestHeaders)
+								for _, check := range testCase.Expectations {
+									check(t, reqInf, resp, tc.Alerts{}, err)
+								}
+							}
+						})
+					case "POST":
+						t.Run(name, func(t *testing.T) {
+							alerts, reqInf, err := testCase.ClientSession.CreateDeliveryServiceRequest(dsReq)
+							for _, check := range testCase.Expectations {
+								check(t, reqInf, nil, alerts, err)
+							}
+						})
+					case "PUT":
+						t.Run(name, func(t *testing.T) {
+							alerts, reqInf, err := testCase.ClientSession.UpdateDeliveryServiceRequestByIDWithHdr(testCase.EndpointId(), dsReq, testCase.RequestHeaders)
+							for _, check := range testCase.Expectations {
+								check(t, reqInf, nil, alerts, err)
+							}
+						})
+					case "DELETE":
```

Comment:
Why is there a case DELETE, when it is already covered in withobjs_test.go?

Reply:
Both the create and delete functions that are used by withobjs are only used for the prerequisites, the only checks those functions perform is that no error occurred, but they do not check status codes or any other validation. And those also do not count as test cases so there is no transparency that post or delete were tested.